crates/core/src/rlp/encode.rs

```diff
@@ -13,6 +13,12 @@ pub trait RLPEncode {
         self.encode(&mut buf);
         buf.len()
     }
+
+    fn encode_to_vec(&self) -> Vec<u8> {
+        let mut buf = Vec::new();
+        self.encode(&mut buf);
+        buf
+    }
```

Comment:
Why not have it as part of the trait? It is more legible and it is a common use case

Reply:
Having it in the trait lets implementors override the default implementation. However, since it's for internal use this won't happen so let's leave it inside the trait.